domains/client/cross-domain-message-gossip/src/gossip_worker.rs

```diff
@@ -28,36 +28,57 @@ pub struct Message {
     pub encoded_data: Vec<u8>,
 }
 
-/// Gossip worker to gossip incoming and outgoing messages to other peers.
-/// Also, streams the decoded extrinsics to destination domain tx pool if available.
-pub struct GossipWorker<Block: BlockT> {
-    gossip_engine: Arc<Mutex<GossipEngine<Block>>>,
-    gossip_validator: Arc<GossipValidator>,
+/// Gossip worker builder
+pub struct GossipWorkerBuilder {
+    gossip_msg_stream: TracingUnboundedReceiver<Message>,
+    gossip_msg_sink: TracingUnboundedSender<Message>,
```

Comment:
Hmm, since `gossip_msg_sink` is ignored when building `GossipWorker`, probably nicer to not include it in the builder, non-blocker.

Reply:
It is for the convenience of the testing framework.